### PR TITLE
Firstboot fixes

### PR DESCRIPTION
**Pull request by keszybz**
Follow-up for #13038.

### CODE REVIEW

**Comment by keszybz** [comment]
`fatal: unable to access 'https://salsa.debian.org/systemd-team/systemd.git/': Received HTTP code 503 from proxy after CONNECT`

File: src/firstboot/firstboot.c [hunk 1]
```diff
@@ -186,33 +186,21 @@ static int prompt_loop(const char *text, char **l, bool (*is_valid)(const char *
...
                         log_info("Selected '%s'.", l[u-1]);
-
-                        c = strdup(l[u-1]);
-                        if (!c)
-                                return log_oom();
-
-                        free(*ret);
-                        *ret = c;
-                        return 0;
+                        return free_and_strdup(ret, l[u-1]);
```
Comment (poettering) [inline src/firstboot/firstboot.c]: this removes error logging on oom

File: src/basic/fileio.c [hunk 2]
```diff
@@ -777,6 +777,7 @@ int read_line_full(FILE *f, size_t limit, ReadLineFlags flags, char **ret) {
         size_t n = 0, allocated = 0, count = 0;
         _cleanup_free_ char *buffer = NULL;
         int r;
+        int tty = -1;
```
Comment (poettering) [inline src/basic/fileio.c]: why not merge with previous line?

Comment (keszybz) [reply src/basic/fileio.c]: Indeed.

File: src/firstboot/firstboot.c (hunk 1 above)
Comment (keszybz) [reply src/firstboot/firstboot.c]: Bummer.

**Pull request merged**